contentcuration/contentcuration/frontend/shared/views/Breadcrumbs.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+
+  <VBreadcrumbs
+    ref="breadcrumbs"
+    v-resize="handleOverflow"
+    :items="items"
+    style="max-height: 100%; width: 100%;"
+  >
+    <template #divider>
+      <Icon>arrow_forward_ios</Icon>
+    </template>
+
+    <!-- Overflow menu -->
+    <VBreadcrumbsItem v-if="breadcrumbStartingIndex > 0" tag="div">
+      <VMenu offset-y bottom>
+        <template #activator="{ on }">
+          <VBtn icon flat class="ma-0" v-on="on">
+            <Icon medium>
+              more_horiz
+            </Icon>
+          </VBtn>
+        </template>
+        <VCard style="max-width: 300px;">
+          <VList v-for="(item, i) in collapsedItems" :key="`collapsed-${i}`">
+            <VListTile :to="item.to">
+              <VListTileTitle>
+                <slot name="item" :item="item" :isLast="false"></slot>
+              </VListTileTitle>
+            </VListTile>
+          </VList>
+        </VCard>
+      </VMenu>
+    </VBreadcrumbsItem>
+
+    <!-- Visible breadcrumbs -->
+    <VBreadcrumbsItem
+      v-for="(item, index) in breadcrumbs"
+      ref="breadcrumb"
+      :key="`breadcrumb-${index}`"
+      tag="div"
+      class="subheading px-2 breadcrumb"
+      :to="(index < breadcrumbs.length - 1)? item.to : undefined"
+    >
+      <slot name="item" :item="item" :isLast="index === breadcrumbs.length - 1"></slot>
+    </VBreadcrumbsItem>
+  </VBreadcrumbs>
+
+</template>
+
+<script>
+
+  export default {
+    name: 'Breadcrumbs',
+    props: {
+      items: {
+        type: Array,
+        default() {
+          return [];
+        },
+        validator(items) {
+          return items.every(i => i.to);
+        },
+      },
+    },
+    data() {
+      return {
+        breadcrumbStartingIndex: 0,
+      };
+    },
+    computed: {
+      collapsedItems() {
+        return this.items.slice(0, this.breadcrumbStartingIndex);
+      },
+      breadcrumbs() {
+        return this.items.slice(this.breadcrumbStartingIndex, this.items.length);
+      },
+    },
+    watch: {
+      items: {
+        handler() {
+          this.handleOverflow();
+        },
+        deep: true,
+      },
+    },
+    methods: {
+      handleOverflow() {
+        let maxWidth = this.$refs.breadcrumbs.$el.offsetWidth;
+
+        let totalWidth = 0;
+        this.breadcrumbStartingIndex = 0;
+        this.$nextTick(() => {
+          for (var i = this.$refs.breadcrumb.length - 1; i >= 0; --i) {
+            totalWidth += this.$refs.breadcrumb[i].$el.offsetWidth + 40;
+
+            // Bounds exceeded, go back to previous index
+            if (totalWidth >= maxWidth - 48) {
+              this.breadcrumbStartingIndex = Math.min(this.items.length - 1, i + 1);
+              break;
+            }
+          }
+        });
+      },
```

Comment:
just a heads up that we have breadcrumbs in Kolibri also with a fair amount of logic of this kind:

https://github.com/learningequality/kolibri-design-system/blob/master/lib/KBreadcrumbs.vue